cborg/src/Codec/CBOR/Read.hs

```diff
@@ -210,11 +212,13 @@ runDecodeAction da = do
 -- of tokens spanning a chunk boundary.
 --
 data SlowPath s a
-   = FastDone               {-# UNPACK #-} !ByteString a
-   | SlowConsumeTokenString {-# UNPACK #-} !ByteString (T.Text     -> ST s (DecodeAction s a)) {-# UNPACK #-} !Int
-   | SlowConsumeTokenBytes  {-# UNPACK #-} !ByteString (ByteString -> ST s (DecodeAction s a)) {-# UNPACK #-} !Int
-   | SlowDecodeAction       {-# UNPACK #-} !ByteString (DecodeAction s a)
-   | SlowFail               {-# UNPACK #-} !ByteString String
+   = FastDone                      {-# UNPACK #-} !ByteString a
+   | SlowConsumeTokenBytes         {-# UNPACK #-} !ByteString (ByteString   -> ST s (DecodeAction s a)) {-# UNPACK #-} !Int
+   | SlowConsumeTokenByteArray     {-# UNPACK #-} !ByteString (BA.ByteArray -> ST s (DecodeAction s a)) {-# UNPACK #-} !Int
```

Comment:
Hmm, do we really have to duplicate this?

Reply:
I don't see how we would construct a `ByteArray#` from a `ByteString` short of copying.